sui/faucet.js

```diff
@@ -9,14 +9,17 @@ async function processCommand(config, chain, options) {
     const [keypair, client] = getWallet(chain, options);
     const recipient = options.recipient || keypair.toSuiAddress();
 
-    await printWalletInfo(keypair, client, chain, options);
+    await printWalletInfo(recipient, client, chain, options);
 
-    await requestSuiFromFaucetV0({
-        host: getFaucetHost(chain.networkType),
-        recipient,
-    });
+    const balance = Number((await client.getBalance({ owner: recipient })).totalBalance) / 1e9;
+    if (balance < Number(options.minBalance)) {
```

Comment:
if balance is >= min, print warning and return instead. otherwise query the faucet

Reply:
done